vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
-HAL_StatusTypeDef mcInit()
-{
-	DEBUG_PRINT("Discharging MCs for 1 second before turning on\n");
-
-    if (mcRightCommand(INVERTER_DISCHARGE_DC_LINK) != HAL_OK) {
-        ERROR_PRINT("Failed to send discharge command to MC Right");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef mcReadParamCommand(uint16_t address, uint16_t data) {
+    VCU_INV_Parameter_RW_Command = INVERTER_PARAM_READ;
+    VCU_INV_Parameter_Address = address;
+    VCU_INV_Parameter_Data = data; // Is this needed for reading? todo
 
-    if (mcLeftCommand(INVERTER_DISCHARGE_DC_LINK) != HAL_OK) {
-        ERROR_PRINT("Failed to send discharge command to MC Left");
+    if (sendCAN_MC_Read_Write_Param_Command() != HAL_OK) {
+        ERROR_PRINT("Failed to send read param message to MC\n");
         return HAL_ERROR;
     }
+    return HAL_OK;
+}
 
-    vTaskDelay(pdMS_TO_TICKS(MC_INIT_DISCHARGE_TIME_MS));
+HAL_StatusTypeDef mcWriteParamCommand(uint16_t address, uint16_t data) {
+    VCU_INV_Parameter_RW_Command = INVERTER_PARAM_WRITE;
+    VCU_INV_Parameter_Address = address;
+    VCU_INV_Parameter_Data = data;
 
-    if (mcRightCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Right");
+    if (sendCAN_MC_Read_Write_Param_Command() != HAL_OK) {
+        ERROR_PRINT("Failed to send write param message to MC\n");
         return HAL_ERROR;
     }
+    return HAL_OK;
+}
 
-    if (mcLeftCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Left");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef mcClearFaults() {
+    return mcWriteParamCommand(INVERTER_FAULT_CLEAR_ADDRESS, 0);
+}
 
+HAL_StatusTypeDef mcInit() {
     DEBUG_PRINT("Waiting for MC to complete startup checks\n");
+    // Todo - Read and check EEPROM values
 
-    // TODO: Do we need to check the voltage values from the MC, or are the BMU
-    // startup checks sufficient?
-
-    DEBUG_PRINT("Starting MC Left...\n");
+    DEBUG_PRINT("Starting MC ..\n");
 
     uint32_t startTick = xTaskGetTickCount();
-    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) &&
-           ((StateInverterLeft & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED))
+
+    // Attempt to disable the lockout
+    while ((xTaskGetTickCount() - startTick < (INVERTER_ON_TIMEOUT_MS)) && 
+           (isLockoutDisabled() == false))
     {
-        sendThrottleValueToMCs(0, 0);
+        sendLockoutReleaseToMC();
+        // mcClearFaults();
         vTaskDelay(pdMS_TO_TICKS(THROTTLE_POLL_TIME_MS));
     }
 
-    if ((StateInverterLeft & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED) {
-        ERROR_PRINT("Timeout waiting for MC Left to be ready to turn on\n");
-        ERROR_PRINT("StateInverterLeft %d\n", (uint8_t)(StateInverterLeft & INVERTER_STATE_MASK));
+
+    if ((isLockoutDisabled() == false) ||
+        (getInverterVSMState() == INV_VSM_State_FAULT_STATE)) {
```

Comment:
Whats VSM? vehicle state machine?

Reply:
yes that is a term from the cascadia docs. itis one of the inverters internal state